addons/utm/models/utm_medium.py

```diff
@@ -20,10 +21,14 @@ class UtmMedium(models.Model):
 
     @api.model_create_multi
     def create(self, vals_list):
-        new_names = self.env['utm.mixin']._get_unique_names(self._name, [vals.get('name') for vals in vals_list])
-        for vals, new_name in zip(vals_list, new_names):
+        seach_domain = expression.OR([[('name', '=ilike', name['name'].lower())] for name in vals_list])
+        existing_records, new_vals_list = self.env['utm.mixin']._find_or_create_lower_record('utm.medium', vals_list, seach_domain)
```

Comment:
> augment it inside _find_or_create_lower_record
why would i need to augment that domain? domain doesn't need to change at all, you just pass domain and that's it.

and i don't think i can use `_find_or_create_record` because it's very different from my goal, it doesn't work on case insensitive manner.

@reth-odoo